test/worker_test.py

```diff
@@ -1296,3 +1298,358 @@ def store_task(t, error_msg):
         w._handle_next_task()
 
         self.assertEqual(result, [task])
+
+
+class PerTaskRetryPolicyBehaviorTest(unittest.TestCase):
```

Comment:
Inherit form `LuigiTestCase` instead of  `unittest.TestCase`, it will handle the cleanup of the classes that you've introduced in every test case.